### PR TITLE
More readable YAML syntax

### PR DESCRIPTION
**Pull request by paddyroddy**
Suggestion to split up #207. To be merged after #210, #212.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,15 +6,16 @@ repos:
...
         exclude: "{{cookiecutter.project_slug}}/src/{{cookiecutter.package_name}}"
       - id: ruff-format
-        args: ["--config", "pyproject.toml"]
+        args:
+          - --config=pyproject.toml
   - repo: https://github.com/codespell-project/codespell
     rev: v2.2.6
     hooks:
       - id: codespell
-        args: [--write-changes]
```
Comment (paddyroddy) [inline .pre-commit-config.yaml]: See #210

File: {{cookiecutter.project_slug}}/.pre-commit-config.yaml [hunk 2]
```diff
@@ -16,7 +16,6 @@ repos:
     rev: v1.6.1
     hooks:
       - id: mypy
-        additional_dependencies: [numpy, pydantic]
```
Comment (paddyroddy) [inline {{cookiecutter.project_slug}}/.pre-commit-config.yaml]: See #212

**Review by samcunliffe — APPROVED** [review]
🤷 Meh. 

I'm fine with this, but I actually find the single-line list more readable.

**Review by dstansby — COMMENTED** [review]
Yeah, I also find the single line (before) more readable

**Comment by dstansby** [comment]
So unless there are others with a strong preference for this PR, I'd suggest we don't merge?

**Comment by paddyroddy** [comment]
Think beyond the one line case.

In this example I find this a lot more readable than the alternative of `["--exclude", ".github", "--warn-list", "yaml[indentation]"` - bearing in mind that each of these can be a list. I want to use this format as I believe it's more extendable, and the `=` make it really obvious that arg2 is tied to arg1
https://github.com/UCL-MIRSG/.github/blob/de59482693728701ceb43ad8f8f9ca4b03c26bc2/precommit/mirsg-hooks.yaml#L32-L38

**Pull request merged**